tests/evaluator_basic_test.py

```diff
@@ -113,31 +107,28 @@ def test_basic_run_task(self):
             ]
             mock_dataset_loader = MagicMock()
             mock_dataset_loader.get_queries_size.return_value = 2
-            mock_dataset_loader.get_queries_for_task.side_effect = (
-                lambda batch_size: iter(
-                    [
-                        {
-                            "query_id": [1, 2],
-                            "query": ["Test query", "Test query 2"],
-                            "answer": ["Test answer", "Test answer 2"],
-                            "table_id": ["Table1", "Table5"],
-                            "database_id": [0, 0],
-                        }
-                    ],
-                )
+            mock_dataset_loader.get_queries_for_task.side_effect = lambda batch_size: iter(
+                [
+                    {
+                        "query_id": [1, 2],
+                        "query": ["Test query", "Test query 2"],
+                        "answer": ["Test answer", "Test answer 2"],
+                        "table_id": ["Table1", "Table5"],
+                        "database_id": [0, 0],
+                    }
+                ],
             )
             mock_func.return_value = {"fetaqa": mock_dataset_loader}
             results = self.evaluator.run(mock_retriever, top_k=2)
             self.assertIsInstance(results, dict)
             self.assertEqual(list(results.keys()), ["Table Retrieval Task"])
-            expected_vals = {"k": 2, "accuracy": 0.5, "precision": None, "recall": None}
-            actual_vals = results["Table Retrieval Task"][
-                "fetaqa"
-            ].retrieval_performance.model_dump()
+            expected_vals = {"k": 2, "accuracy": 0.5, "precision": None, "recall": 0.5}
+            actual_vals = results["Table Retrieval Task"]["fetaqa"].retrieval_performance.model_dump()
 
             for key, value in expected_vals.items():
                 self.assertIn(key, actual_vals)
                 self.assertEqual(value, actual_vals[key])
```

Comment:
Fails with `FAILED tests/evaluators/evaluator_basic_test.py::TestEvaluator::test_basic_run_task - AssertionError: 0.5 != 0.0`

Reply:
resolved

Reply:
@jixy2012 This is still failing for me when invoking via `pytest tests/ --ignore=tests/tasks/fact_ver_task_basic_test.py --ignore=tests/tasks/nih_basic_test.py` (to ignore the tests that take 2+ minutes to run), and also via your direct invocation (`python tests/evaluators/evaluator_basic_test.py`)

This is the full error I get: 
```
FAIL: test_basic_run_task (__main__.TestEvaluator)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/parkerglenn/Desktop/projects/target/tests/evaluators/evaluator_basic_test.py", line 135, in test_basic_run_task
    self.assertEqual(value, actual_vals[key])
AssertionError: 0.5 != 0.0

======================================================================
FAIL: test_dataset_loaders_loading (__main__.TestEvaluator)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/parkerglenn/Desktop/projects/target/tests/evaluators/evaluator_basic_test.py", line 85, in test_dataset_loaders_loading
    self.assertTrue(expected.items() <= feta_loader.corpus[0].items())
AssertionError: False is not true
```

Same with the assert on accuracy mentioned above in `table_retrieval_task_basic_test.py`, any ideas?

Reply:
I don't actually know what's trying to be tested via the line `self.assertTrue(expected.items() <= feta_loader.corpus[0].items())`, it looks like this might be missing a `len()`?